modules/web/server/src/main/scala/engage/web/server/http4s/GraphQlRoutes.scala

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package engage.web.server.http4s
+
+import cats.effect.Async
+import cats.syntax.all.*
+import engage.model.security.UserDetails
+import engage.server.EngageEngine
+import engage.web.server.security.AuthenticationService
+import engage.web.server.security.Http4sAuthentication
+import engage.web.server.security.TokenRefresher
+import lucuma.graphql.routes.GrackleGraphQLService
+import lucuma.graphql.routes.Routes
+import natchez.Trace
+import org.http4s.AuthedRoutes
+import org.http4s.HttpRoutes
+import org.http4s.dsl.Http4sDsl
+import org.http4s.server.middleware.GZip
+import org.http4s.server.websocket.WebSocketBuilder2
+import org.typelevel.log4cats.Logger
+
+class GraphQlRoutes[F[_]: Async: Logger: Trace](
+  eng: EngageEngine[F]
+) extends Http4sDsl[F] {
+
+  private def commandServices(wsb: WebSocketBuilder2[F]): HttpRoutes[F] = GZip(
+    Routes.forService(_ => EngageMappings(eng).map(GrackleGraphQLService[F](_).some), wsb, "engage")
```

Comment:
I couldn't figure out how to use `AuthRoutes` here.

Reply:
No `AuthedRoutes` here,  use lucuma-sso-client to do the authentication. Here is how the odb does it: https://github.com/gemini-hlsw/lucuma-odb/blob/main/modules/service/src/main/scala/lucuma/odb/graphql/GraphQLRoutes.scala

Reply:
It is not a given that engage will use Lucuma-sso, it is for internal users authenticating with noirlab sso
I guess it is open if we'll allow login via orcid